### PR TITLE
Add CD & PD to concepts page

### PR DESCRIPTION
**Pull request by FavourDaniel**
Added new concepts (continuous delivery, continuous deployment, and progressive delivery) to resolve issue #783

Fix: #783

### CODE REVIEW

File: content/en/flux/concepts.md [hunk 1]
```diff
@@ -71,3 +71,35 @@ The bootstrap is done using the `flux` CLI or
...
+updates to users as the approach allows developers to focus on writing code, while automated processes 
+handle the tasks of testing and deploying the code changes.
+
+## Progressive Delivery
+
+Progressive Delivery builds on Continuous Delivery by gradually rolling out new features or updates 
+to a subset of users, allowing developers to test and monitor the new features in a controlled environment 
+and make necessary adjustments before releasing them to everyone.
+
+Developers can use techniques like feature flags, canary releases, and A/B testing to minimize the chances
```
Comment (stefanprodan) [inline content/en/flux/concepts.md]: Let's add links to https://glossary.cncf.io/canary-deployment/  and https://fluxcd.io/flagger/usage/deployment-strategies/

File: content/en/flux/concepts.md [hunk 2]
```diff
@@ -71,3 +71,35 @@ The bootstrap is done using the `flux` CLI or
...
 
 For more information, take a look at [the bootstrap documentation](installation.md#bootstrap).
+
+## Continuous Delivery
+
+Continuous Delivery is a software development practice that automates the software release 
+process up to the point of deployment except that manual intervention is required at some point. 
+As code changes are made, they undergo testing, and artifacts are subsequently created. 
+These artifacts are deployed to a staging environment for additional manual testing and approval 
+before being released to production servers.
```
[the remark this answers is not in the document]
Comment (FavourDaniel) [reply content/en/flux/concepts.md:79]: Alright then

File: content/en/flux/concepts.md [hunk 3]
```diff
@@ -73,22 +73,19 @@ using our [Terraform Provider](https://github.com/fluxcd/terraform-provider-flux
...
-Continuous Delivery is a software development practice that automates the software release process up to the point of deployment except that manual intervention is required at some point. As code changes are made, they undergo testing, and artifacts are subsequently created. 
-These artifacts are deployed to a staging environment for additional manual testing and approval 
-before being released to production servers.
-
-By adopting this practice, software teams can update their code more frequently and reliably with 
-less risk, while maintaining control over deployment and improving overall software quality.
+For more information, take a look at continuous delivery as defined by the [CNCF](https://glossary.cncf.io/continuous-delivery/).
 
 ## Continuous Deployment
+Continuous Deployment is the practice of automatically deploying code changes to production once they have passed through automated testing.
```
Comment (stefanprodan) [inline content/en/flux/concepts.md]:
```suggestion
## Continuous Deployment

Continuous Deployment is the practice of automatically deploying code changes to production once they have passed through automated testing. 
```

File: content/en/flux/concepts.md [hunk 4]
```diff
@@ -73,22 +73,19 @@ using our [Terraform Provider](https://github.com/fluxcd/terraform-provider-flux
 For more information, take a look at [the bootstrap documentation](installation.md#bootstrap).
 
 ## Continuous Delivery
+Continuous Delivery refers to the practice of delivering software updates frequently and reliably.
```
Comment (stefanprodan) [inline content/en/flux/concepts.md]:
```suggestion
## Continuous Delivery

Continuous Delivery refers to the practice of delivering software updates frequently and reliably. 
```

File: content/en/flux/concepts.md [hunk 5]
```diff
@@ -73,22 +73,19 @@ using our [Terraform Provider](https://github.com/fluxcd/terraform-provider-flux
...
-
-As code changes are made, they are automatically built, tested, and deployed to production servers without human or manual intervention. This significantly reduces the time it takes to deliver software updates to users as the approach allows developers to focus on writing code, while automated processes handle the tasks of testing and deploying the code changes.
+For more information, take a look at continuous deployment as defined by the [CNCF](https://glossary.cncf.io/continuous-delivery/).
 
 ## Progressive Delivery
 
 Progressive Delivery builds on Continuous Delivery by gradually rolling out new features or updates to a subset of users, allowing developers to test and monitor the new features in a controlled environment and make necessary adjustments before releasing them to everyone.
 
 Developers can use techniques like feature flags, canary releases, and A/B testing to minimize the chances of introducing bugs or errors that could harm users or interrupt business operations. These strategies enable a controlled and gradual rollout of new features, ensuring a smooth and successful release that enhances user trust and improves the overall user experience.
```
Comment (stefanprodan) [inline content/en/flux/concepts.md:89]:
```suggestion
Developers can use techniques like feature flags, [canary releases](https://glossary.cncf.io/canary-deployment/), and A/B testing to minimize the chances of introducing bugs or errors that could harm users or interrupt business operations. These strategies enable a controlled and gradual rollout of new features, ensuring a smooth and successful release that enhances user trust and improves the overall user experience.
```

**Comment by FavourDaniel** [comment]
@stefanprodan I have implemented the changes, please take a look at it, thank you.

File: content/en/flux/concepts.md [hunk 6]
```diff
@@ -73,22 +73,19 @@ using our [Terraform Provider](https://github.com/fluxcd/terraform-provider-flux
...
+
+For more information, take a look at [canary deployment](https://glossary.cncf.io/canary-deployment/) and [deployment strategies](https://fluxcd.io/flagger/usage/deployment-strategies/) in Flux.
```
Comment (stefanprodan) [inline content/en/flux/concepts.md:91]:
```suggestion
The Flux project offers a specialised controller called [Flagger](https://github.com/fluxcd/flagger) that implements various progressive delivery techniques. For more information, take a look at [Flagger deployment strategies](https://fluxcd.io/flagger/usage/deployment-strategies/).
```

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @FavourDaniel 🏅 

PS. Can you please squash all commits into a single one.

**Comment by FavourDaniel** [comment]
@stefanprodan done

**Comment by stefanprodan** [comment]
Awesome! Thanks a lot @FavourDaniel for this contribution to Flux docs.

**Comment by FavourDaniel** [comment]
Thank you so much @stefanprodan, I will definitely pick up more issues to contribute to.

**Pull request merged**